Scav Randomizer V2

BALIST0N opened the pull request:
#### Randomized bots !
the server now generate randomized AI scavs, only based on RNG

##### What is randomized ?
 
- Appearance : head and outfit are mixed with all kind of scavs( bosses, pmcs etcc..)
- weapons : more than 160 possibilities including pistols , grenades & ultra modded weapons
- gear : scavs can have any rigs, armor, backpack and headgear , each slot is based on % of chance
- skills : all their skills are randomized (endurance, strengh, vitality, etcc)

#####  randomization 
Every bot randomized have these following chances of stuff :

- Vest rig : 100% (can be armored, they are mixed with normal rigs)
- Magazines : 100% , randomized magazine on his weapon and two on his rig
- Ammo : 100% : stacks of ammo on each mags are randomized
- Glasses : 30%
- Face cover : 40%
- headwear : 40% (it can be an armored Helmet, but no visor...)
- Backpack : 25%
- Body armor : 25% 
- Grenade in pocket : 10% (except bosses)
- meds in pocket : 10% (except bosses)
- Snipers scavs can only have DMR, carabines, or snipers rifles 

#### Scav bosses ?
Killa and Reshala are vanillia, but reshala's guards stuff is randomized, their behavior are still the same. So be very carefull when hunting bosses

#### How it works ?
the server take care of all !  Nothing to configure, nothing to edit : )

##### PmcWar Mode
the randomizer provide a pmc war mode, it replace randomly every scav by a bear or a usec
witch lead into some epic team fight with AI

therefore, this setting is hard on CPU usage and make the game sluttering ..
there is one option to change in "bots/BotsSettings.json"

    "EnablePmcWar":true,
(its false by default)


________________________________________

PS : I made those change you wanted, easy to tweak if needed
I made a lot of changes but the idea is still the same

in server.JS the only change are two things : the "generateBots" function (  [line103](https://github.com/polivilas/EmuTarkov/pull/12/files#diff-78c12f5adc1848d13b1c6f07055d996eR103) )
and a case on "handleRequest" [line 824](https://github.com/polivilas/EmuTarkov/pull/12/files#diff-78c12f5adc1848d13b1c6f07055d996eR824)...

BALIST0N committed:
Rollback and reupdate

(messed up files...))

BALIST0N committed:
Add files via upload

BALIST0N committed:
Scav randomizer update 5

- added settings file , it maybe usefull later
- added "PMC WAR MODE" , instead of generating scavs, the randomizer generate both Usec and Bear AI, witch lead into some epic AI fights

BALIST0N committed:
Delete bot_generate_v2.json

BALIST0N committed:
Bot generating updates

- killa and reshala are reverted to vanilla , but not reshala's guards
- some changes I don't remember , I messed up so much ….

BALIST0N committed:
Update README.md

BALIST0N committed:
clear profile and disabled devlopper mode 

just fucking enjoy

BALIST0N committed:
Update README.md

BALIST0N committed:
Update README.md

BALIST0N committed:
marksman class into the scav randomization

sniper scavs can only have some carabines dmr or sniper preset : )

BALIST0N committed:
reupdate readme

polivilas reviewed (commented):
Is this intentional? I don't think PMC has 70 head health.

polivilas commented on list.json:
```diff
@@ -721,44 +581,44 @@
 						"Health": {
-							"Current": 35,
-							"Maximum": 35
+							"Current": 70,
```
This is the change in quesiton.

BALIST0N replied:
oops totally forgot to revert normal health for the profile
I will update in a minute : )

BALIST0N committed:
reverted normal health for character

BALIST0N committed:
Update list.json

InNoHurryToCode commented:
Ran this version of the server for a couple of hours  (~3-4) on various ingame AI settings without any issues. Didn't test the PMC war through.

BALIST0N committed:
added result of research of various things... 

I made research about quests management (progressions, completing, accepting and rewarding)
the aim is to create very custom quests for emutarkov
also about healing and eating/drinking consumable management server side

but all my research need extra help, also code is in comment because its not fully working 

edited things are in "handlemoving" function in server.js 
case "QuestComplete"
case "QuestHandover"
case "Heal"
case "Eat"

about flea market I'm still hoping someone will give me the help I need to finish this thing …

polivilas commented:
:+1:, no bad behavior on my side, thank you!
